docs/coding-guidelines/add-types-to-SerDe.md

```diff
@@ -0,0 +1,155 @@
+    private static readonly byte[] s_stringTypeId = new[] { (byte)'c' };
+    private static readonly byte[] s_boolTypeId = new[] { (byte)'b' };
+    private static readonly byte[] s_doubleTypeId = new[] { (byte)'d' };
+    private static readonly byte[] s_jvmObjectTypeId = new[] { (byte)'j' };
+    private static readonly byte[] s_byteArrayTypeId = new[] { (byte)'r' };
+    private static readonly byte[] s_arrayTypeId = new[] { (byte)'l' };
+    private static readonly byte[] s_dictionaryTypeId = new[] { (byte)'e' };
+    private static readonly byte[] s_rowArrTypeId = new[] { (byte)'R' };
+
+```
+
+Once you have created your new identifier implement the csharp side writer in PayloadHelper.cs by 
+finding the method `ConvertArgsToBytes` and find out where in the switch that the code should be 
+implemented.
+
+This is how a byte array is written:
+
+```csharp
+
+    if (addTypeIdPrefix)
+    {
+        SerDe.Write(destination, GetTypeId(argType));
+    }
+
+...
+
+    case byte[] argByteArray:
+        SerDe.Write(destination, argByteArray.Length);
+        SerDe.Write(destination, argByteArray);
+        break;
+
+```
+
+Typically the type identifier is written, then the length and then the data. Once you have written 
+the code that will perform the write you need to add a case to `GetTypeId` for your new identifier:
+
+```csharp
+
+    case TypeCode.Double:
+        return s_doubleTypeId;
+
+```
+
+The next step is to add the derserialization code to scala (read code).
+
+In src/main/scala/org/apache/spark/api/dotnet/SerDe.scala (These are versioned so you may need to 
+try and find it). In the method `readTypedObject` add a new case statement for your new type 
+identifier:
+
+```scala
+
+def readTypedObject(dis: DataInputStream, dataType: Char): Object = {
+    dataType match {
+      case 'n' => null
+      case 'i' => new java.lang.Integer(readInt(dis))
+      case 'g' => new java.lang.Long(readLong(dis))
+      case 'd' => new java.lang.Double(readDouble(dis))
+      case 'b' => new java.lang.Boolean(readBoolean(dis))
+      case 'c' => readString(dis)
+      case 'e' => readMap(dis)
+      case 'r' => readBytes(dis)
+      case 'l' => readList(dis)
+      case 'D' => readDate(dis)
+      case 't' => readTime(dis)
+      case 'j' => JVMObjectTracker.getObject(readString(dis))
+      case 'R' => readRowArr(dis)
+      case _ => throw new IllegalArgumentException(s"Invalid type $dataType")
+    }
+  }
+
+```
+
+Once you have read the type, you can implement your method to read the data you wrote in csharp, examples:
+
+```scala
+
+  def readBoolean(in: DataInputStream): Boolean = {
+    in.readBoolean()
+  }
+
+  def readRow(in: DataInputStream): Row = {
+    val len = readInt(in)
+    Row.fromSeq((0 until len).map(_ => readObject(in)))
+  }
+
+  def readBytesArr(in: DataInputStream): Array[Array[Byte]] = {
+    val len = readInt(in)
+    (0 until len).map(_ => readBytes(in)).toArray
+  }
+
+```
```

Comment:
Can you please show one example to be consistent with E2E example? e.g. `DoubleArrayArray`.